### DISCUSSION TITLE
local mode has been disabled by your administrator

### DISCUSSION THREAD

**Discussion by G0nz0uk** [grafana/grafana-csv-datasource#97] [Q&A] [answered]
Hello,

I've had to restart our Grafana docker today and for some reason I get:

**"local mode has been disabled by your administrator"**

If I go to my datasource and test it says it's working.

![image](https://user-images.githubusercontent.com/15994497/123433783-b9f34900-d5c3-11eb-88e7-ce3b05714511.png)

![image](https://user-images.githubusercontent.com/15994497/123434013-f1fa8c00-d5c3-11eb-8bbd-cc37a76dbae8.png)

Using Grafana 8.0.1 and Sign CSV 0.6.1

Any help would be great!

**Comment by marcusolsson** [comment, marked as answer]
v0.6.0 was a security update. Allowing local mode means that Grafana users can read files on the local machine, which can pose security risks if there's sensitive information on the same file system as the Grafana server.

For this reason I've blocked local mode by default. If you want to enable it, please see the documentation:

https://marcus.se.net/grafana-csv-datasource/configuration#allow-local-mode

**Reply by G0nz0uk** [reply]
Hi Marcus,  I had to move all mine to local mode as IIS wouldn't work anymore for me as a remote source, you might remember all the testing I did and we couldn't figure it out.

How do I test this security issue then?  My Grafana users are AD authenticated and Grafana is on a Linux server so I'm not sure how they would access the local files with their credentials.

It would be Good to know how to test.

**Reply by marcusolsson** [reply]
Backend plugin (like this one) have the same permissions as the grafana-server process, which means that if you configure the local path in the data source config to something like `/`, viewers would be able to read for example the grafana.ini file if they knew where it was located.

You can test it by enabling local mode, setting the path to `/`, and in the query do something like `etc/secret.json`, to see if they can access it.

**Reply by G0nz0uk** [reply]
I will try that thanks.

I'm struggling to enable it as we use docker and and environment file for custom attributes for the grafana.ini

for example for SNMP I use this:
GF_INSTALL_PLUGINS=alexanderzobnin-zabbix-app,grafana-clock-panel,marcusolsson-csv-datasource
GF_SMTP_ENABLED=true
GF_SMTP_HOST=10.10.10.123:25
GF_SMTP_FROM_ADDRESS=grafana-monitor@domain.com
GF_SMTP_FROM_NAME=Grafana-Monitor

How would yours look, I did try the below with no success:

GF_PLUGIN.MARCUSOLSSON-CSV-DATASOURCE_ALLOW_LOCAL_MODE=true

**Reply by marcusolsson** [reply]
Hm. I haven't tried it with environment variables, but if you set it in the grafana.ini, it's actually exposed as a variable:

https://github.com/marcusolsson/grafana-csv-datasource/blob/85aa0a33af3a6a0f728eaae8dde3ffe14922b67b/pkg/datasource.go#L47

Does it work if you set:

`GF_PLUGIN_ALLOW_LOCAL_MODE=true`?

**Reply by G0nz0uk** [reply]
Yes that worked, thanks.

To test the security issue do you mean create a datasource to /, and query something like etc/secret.json?
Is the query in a new panel?

**Reply by mikethebee** [reply]
My Grafana in running on a raspberryPi on an intranet and I need local mode. Instructions at https://marcus.se.net/grafana-csv-datasource/configuration#allow-local-mode work, but it would help noobs to suggest the default location for the configuration on linux is /etc/grafana/grafana.ini

**Reply by ChlaegerIO** [reply]
In my case I could solve it with the following two lines in the `grafana.ini` file. 
```
[plugin.marcusolsson-csv-datasource]
allow_local_mode = true
```
See also the redit source: https://www.reddit.com/r/grafana/comments/o8ac4f/csv_plugin_local_mode_has_been_disabled_by_your/.

**Reply by spr-01** [reply]
I have added the below to the Custom.ini, restarted Grafana and yet I still get the same error.

Any ideas?

I am using Windows.

[plugin.marcusolsson-csv-datasource]
allow_local_mode = true

**Comment by diogobaltazar** [comment]
I can enable local mode with before launching the server by:

```shell
export GF_PLUGIN_ALLOW_LOCAL_MODE=true
```

But I can't enable local mode with `grafana.ini`. Where do you place this file? I've tried with no avail to:

+ `/opt/homebrew/etc/grafana/grafana.ini`
+ `/usr/local/etc/grafana/grafana.ini`

with contents:

```ini
[plugin.marcusolsson-csv-datasource]
allow_local_mode = true
```

**Reply by Krishanadave671** [reply]
how did you export i didn't understood ?
